src/Workspaces/Core/Portable/Workspace/ProjectSystem/FileWatchedPortableExecutableReferenceFactory.cs

```diff
@@ -31,7 +31,7 @@ internal sealed class FileWatchedReferenceFactory<TReference>
     /// are only created once we are actually applying a batch because we don't determine until the batch is applied if
     /// the file reference will actually be a file reference or it'll be a converted project reference.
     /// </summary>
-    private readonly Dictionary<TReference, (IWatchedFile Token, int RefCount)> _referenceFileWatchingTokens = [];
+    private readonly Dictionary<string, (IWatchedFile Token, int RefCount)> _referenceFileWatchingTokens = [];
```

Comment:
this is the crux of the change.  it keeps the token/ref-count assocaited with the path, not the reerence identity anymore.

Reply:
Dumb question: Could _previousDisposalLocations key on the path too?